File: problems/1160-find-words-that-can-be-formed-by-characters/solution.java
class Solution {
    public int countCharacters(String[] words, String chars) {
        int[] charCounts = new int[26];
        for (char c : chars.toCharArray()) {
            charCounts[c - 'a']++;
        }

        int totalLength = 0;
        for (String word : words) {
            int[] wordCounts = new int[26];
            boolean good = true;
            for (char c : word.toCharArray()) {
                int index = c - 'a';
                wordCounts[index]++;
                if (wordCounts[index] > charCounts[index]) {
                    good = false;
                    break;
                }
            }
            if (good) {
                totalLength += word.length();
            }
        }

        return totalLength;
    }
}

File: problems/1160-find-words-that-can-be-formed-by-characters/solution.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        char_counts = {}
        for char in chars:
            char_counts[char] = char_counts.get(char, 0) + 1
        
        total_length = 0
        for word in words:
            word_counts = {}
            is_good = True
            for char in word:
                word_counts[char] = word_counts.get(char, 0) + 1
                if char not in char_counts or word_counts[char] > char_counts[char]:
                    is_good = False
                    break
            
            if is_good:
                total_length += len(word)
        
        return total_length